### ISSUE TITLE
Multiple cors_rule [FEAT] ...

### ISSUE THREAD

**Issue by sergio91arreglatech** [claranet/terraform-azurerm-storage-account#9]
### Community Note

* Please vote on this issue by adding a :thumbsup: [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
* Please do not leave "+1" or "me too" comments, they generate extra noise for issue followers and do not help prioritize the request
* If you are interested in working on this issue or have submitted a pull request, please leave a comment

### Description

Create mutiple cors rules inside blob_properties

### New or Affected Resource(s)/Data Source(s)

azurerm_storage_account

### Potential Terraform Configuration

```hcl
variable "storage_blob_cors_rule" {
  description = "A map of storage account settings including CORS rules."
  type        = map(object({
    cors_rules = list(object({
      allowed_origins     = list(string)
      allowed_methods     = list(string)
      allowed_headers     = list(string)
      exposed_headers     = list(string)
      max_age_in_seconds  = number
    }))
  }))
  default = {
    default = {
      cors_rules = [
        {
          allowed_origins     = ["*"]
          allowed_methods     = ["GET"]
          allowed_headers     = ["*"]
          exposed_headers     = ["*"]
          max_age_in_seconds  = 3600
        },
      ]
    }
  }
}
```

**Comment by Shr3ps** [comment]
Hello @sergio91arreglatech , thanks for this suggestion.

I think this variable type should be enough:
```terraform
variable "storage_blob_cors_rule" {
  description = "A list of storage account blob CORS rules."
  type        = list(object({
      allowed_origins     = list(string)
      allowed_methods     = list(string)
      allowed_headers     = list(string)
      exposed_headers     = list(string)
      max_age_in_seconds  = number
    }))
```

Are you able to open a pull-request for implementation? or should we add this to our todo-list?

**Comment by Shr3ps** [comment]
released in v7.12.0
